core/main/src/firebolt/handlers/advertising_rpc.rs

```diff
@@ -162,12 +162,12 @@ impl AdvertisingServer for AdvertisingImpl {
         let payload = AdvertisingRequest::GetAdIdObject(AdIdRequestParams {
             privacy_data: privacy_rpc::get_allow_app_content_ad_targeting_settings(&self.state)
                 .await,
-            app_id: ctx.clone().app_id.to_string(),
+            app_id: ctx.clone().app_id,
```

Comment:
Rather than cloning the ctx to get app_id, I think we can do ctx.app_id.to_string() or ctx.app_id.to_owned()